From d0575953ec8a57442a901b94cbad4d512407202e Mon Sep 17 00:00:00 2001
From: SUGIOKA Toshinobu <sugioka@itonet.co.jp>
Date: Mon, 10 Aug 2009 08:12:49 -0700
Subject: [PATCH 072/100] SH: fix clobber list in stdlib/longlong.h

(cherry picked from commit 78fd882a7bf6153f33b73363c542e1f37c098c56)
---
 ChangeLog         |    5 +++++
 stdlib/longlong.h |    7 ++++---
 2 files changed, 9 insertions(+), 3 deletions(-)

2009-08-10  SUGIOKA Toshinobu  <sugioka@itonet.co.jp>

	* stdlib/longlong.h [__sh__] (udiv_qrnnd, sub_ddmmss): Add "t" to
	clobber list.

diff --git a/stdlib/longlong.h b/stdlib/longlong.h
index a2f38ae..e7d6099 100644
--- a/stdlib/longlong.h
+++ b/stdlib/longlong.h
@@ -1,6 +1,6 @@
 /* longlong.h -- definitions for mixed size 32/64 bit arithmetic.
    Copyright (C) 1991, 1992, 1994, 1995, 1996, 1997, 1998, 1999, 2000,
-   2002, 2003, 2004, 2005, 2006 Free Software Foundation, Inc.
+   2002, 2003, 2004, 2005, 2006, 2009 Free Software Foundation, Inc.
 
    This file is part of the GNU C Library.
 
@@ -918,7 +918,7 @@ UDItype __umulsidi3 (USItype, USItype);
 "	or r1,%0"							\
 	: "=r" (q), "=&z" (r)						\
 	: "1" (n1), "r" (n0), "rm" (d), "r" (&__udiv_qrnnd_16)		\
-	: "r1", "r2", "r4", "r5", "r6", "pr");				\
+	: "r1", "r2", "r4", "r5", "r6", "pr", "t");			\
   } while (0)
 
 #define UDIV_TIME 80
@@ -926,7 +926,8 @@ UDItype __umulsidi3 (USItype, USItype);
 #define sub_ddmmss(sh, sl, ah, al, bh, bl)				\
   __asm__ ("clrt;subc %5,%1; subc %4,%0"				\
 	   : "=r" (sh), "=r" (sl)					\
-	   : "0" (ah), "1" (al), "r" (bh), "r" (bl))
+	   : "0" (ah), "1" (al), "r" (bh), "r" (bl)			\
+	   : "t")
 
 #endif /* __sh__ */
 
-- 
1.6.5.2

